src/ebonite/build/helpers.py

```diff
@@ -1,12 +1,19 @@
 from functools import wraps
-from typing import List
+from typing import List, Callable
 from pyjackson.utils import get_function_fields, turn_args_to_kwargs
 
 
-def validate_kwargs(f=None, *, allowed: List[str] = None):
+def validate_kwargs(f=None, *, allowed: List[str] = None, allowed_funcs: List[Callable] = None):
 
     def inner(func):
-        all_fields = set(allowed or [])
+        all_fields = allowed or []
+        if allowed_funcs is not None:
+            from_funcs = []
+            [from_funcs.extend(get_function_fields(allowed_func, False)) for allowed_func in allowed_funcs]
+            all_fields += [field.name for field in from_funcs]
+            if 'self' in all_fields:
```

Comment:
check if 'self' really there. if i remember correctly, get_function_fields may drop it already

Reply:
will look into it